query/define.rs

```diff
@@ -363,151 +516,442 @@ fn define_capabilities_relates(
             return Err(err_unsupported_capability(&label, type_.kind(), capability));
         };
 
-        let (role_label, ordering) = type_ref_to_label_and_ordering(&relates.related).map_err(|_| {
-            DefineError::RelatesRoleMustBeLabelAndNotOptional {
-                relation: label.to_owned(),
-                role_label: relates.related.clone(),
-            }
-        })?;
-        let existing_opt = relation_type
-            .get_relates_role_name(snapshot, type_manager, role_label.name.as_str())
-            .map_err(|source| DefineError::UnexpectedConceptRead { source })?;
-        let created = if let Some(existing_relates) = existing_opt {
-            let existing_ordering = existing_relates
-                .role()
-                .get_ordering(snapshot, type_manager)
+        let (role_label, ordering) = type_ref_to_label_and_ordering(&label, &relates.related)
+            .map_err(|source| DefineError::DefinitionResolution { source })?;
+
+        let definition_status =
+            get_relates_status(snapshot, type_manager, relation_type.clone(), &role_label, ordering)
                 .map_err(|source| DefineError::UnexpectedConceptRead { source })?;
-            if ordering != existing_ordering {
-                Err(DefineError::CreateRelatesModifiesExistingOrdering {
+        let (defined, is_new) = match definition_status {
+            DefinitionStatus::DoesNotExist => {
+                let init_cardinality = if let Some(typeql_cardinality) = capability
```

Comment:
Will be removed in the next PR when the init_cardinality is removed from the API!